torch/lib/THD/CMakeLists.txt

```diff
@@ -67,9 +63,8 @@ IF(MPI_FOUND)
   MESSAGE(STATUS "MPI_LIBRARIES: ${MPI_LIBRARIES}")
 ENDIF()
 
-IF(GLOO_FOUND)
+IF(USE_GLOO AND USE_CUDA)
```

Comment:
I wont repeat the THD comments from #12775 again over here, have a look at that review